src/watchdog/observers/inotify.py

```diff
@@ -202,6 +208,32 @@ def _decode_path(self, path):
             return path
         return os.fsdecode(path)
 
+    def get_event_mask_from_filter(self):
+        """Optimization: Only include events we are filtering in inotify call"""
+        if self._event_filter is None:
+            return None
+
+        # always listen to delete self
+        event_mask = InotifyConstants.IN_DELETE_SELF
+        for cls in self._event_filter:
+            if cls in (DirMovedEvent, FileMovedEvent):
+                event_mask |= InotifyConstants.IN_MOVE
```

Comment:
hey @marioga @BoboTiG 
first off, thanks for this very appreciated effort

I have a small question- 
If I run
```python
        observer.schedule(
            ...
            event_filter=[FileMovedEvent],
        )
```
We save a lot of ram & cpu by not having to process events like file deletion, etc.
but since both `file-moved` and `dir-moved` are mapped to a single `inotify` flag `IN_MOVE`
We will still queue both events as part of the event-queue (leading to an undesired memory growth that we can avoid)

wouldn't it be better to add another check on top of the low-level inotify mask
to decide whether or not we should queue the event as part of `queue_event()`?

WDYT?